packages/ifpack2/src/Ifpack2_PowerMethod.hpp

```diff
@@ -152,70 +156,142 @@ powerMethodWithInitGuess (const OperatorType& A,
      "compute the initial guess), or because the norm2 method has a "
      "bug.  The first is not impossible, but unlikely.");
 
-  if (verbose) {
-    *out << "  Original norm1(x): " << x->norm1 ()
+  if(verbose) {
+    *out << "  Original norm1(x): " << x->norm1()
           << ", norm2(x): " << norm << endl;
   }
 
-  x->scale (one / norm);
+  x->scale(one / norm);
 
-  if (verbose) {
-    *out << "  norm1(x.scale(one/norm)): " << x->norm1 () << endl;
+  if(verbose) {
+    *out << "  norm1(x.scale(one/norm)): " << x->norm1() << endl;
   }
 
   if(y.is_null())
-    y = Teuchos::rcp(new V(A.getRangeMap ()));
-
-  for (int iter = 0; iter < numIters-1; ++iter) {
-    if (verbose) {
-      *out << "  Iteration " << iter << endl;
+    y = Teuchos::rcp(new V(A.getRangeMap()));
+
+  // GH 04 Nov 2022: The previous implementation of the power method was inconsistent with itself.
+  // It computed x->norm2() inside the loop, but returned x^T*Ax.
+  // This returned horribly incorrect estimates for Chebyshev spectral radius in certain 
+  // cases, such as the Cheby_belos test, which has two dominant eigenvalues of 12.5839, -10.6639.
```

Comment:
The estimate was correct given the number of iterations.

Reply:
A negative estimate for a spectral radius is incorrect though?

Reply:
The method was computing an estimate for the dominant eigenvalue. After 10 its, it returned the correct (but arguably bad) estimate. After 30 its the estimate was better. The method itself was fine though, so I wouldn't say that something was "horribly incorrect", especially given that your implementation will return the exact same estimate when called with `computeSpectralRadius=true` after 10 iterations.

Reply:
> given that your implementation will return the exact same estimate when called with computeSpectralRadius=true after 10 iterations.

Wait now I'm confused. `computeSpectralRadius=true` will definitely not return the same answer as it did previously for 10 iterations. `computeSpectralRadius=true` returns $\\|D^{-1}Ax\\|_2$ instead of the previous $x^TD^{-1}Ax$.

Reply:
Sorry, `computeSpectralRadius=false`